src/test/kotlin/org/jetbrains/research/ml/ast/util/FileTestUtil.kt

```diff
@@ -2,45 +2,35 @@ package org.jetbrains.research.ml.ast.util
 
 import java.io.File
 
+enum class Extension(val value: String) {
+    Py(".py"), Xml(".xml")
+}
+
 object FileTestUtil {
 
     val File.content: String
         get() = this.readText().removeSuffix("\n")
 
     fun getInAndOutFilesMap(
         folder: String,
-        oldExtension: String = ".py",
-        newExtension: String = ".py"
+        inExt: Extension = Extension.Py,
+        outExt: Extension = Extension.Py
     ): Map<File, File> {
-        val inFileRegEx = "in_\\d*$oldExtension".toRegex()
-        val inOutFileRegEx: Regex = "(in|out)_\\d*($oldExtension|$newExtension)".toRegex()
-        val (inFiles, outFiles) = getNestedFiles(folder).toList().filter { inOutFileRegEx.containsMatchIn(it.name) }
-            .partition { inFileRegEx.containsMatchIn(it.name) }
-        if (inFiles.size != outFiles.size) {
-            throw IllegalArgumentException(
-                "Size of the list of input files does not equal size of the list of output files if the folder: $folder"
-            )
-        }
-        return inFiles.associateWith { inFile ->
-            // TODO: can I do it better?
-            val outFileName = inFile.name.replace("in", "out").replace(oldExtension, newExtension)
-            val outFile = File("${inFile.parent}/$outFileName")
-            if (!outFile.exists()) {
-                throw IllegalArgumentException("Out file $outFile does not exist!")
-            }
-            outFile
-        }
-    }
+        fun File.number(): Int = "(?<=(in|out)_)\\d+".toRegex().find(this.name)!!.value.toInt()
+        fun File.isInFile(): Boolean = "in_\\d+.*${inExt.value}".toRegex().containsMatchIn(this.name)
+        fun File.isOutFile(): Boolean = "out_\\d+.*${outExt.value}".toRegex().containsMatchIn(this.name)
 
-    private fun getNestedFiles(directoryName: String, files: MutableList<File> = ArrayList()): Sequence<File> {
-        val root = File(directoryName)
-        root.listFiles()?.forEach {
-            if (it.isFile) {
-                files.add(it)
-            } else if (it.isDirectory) {
-                getNestedFiles(it.absolutePath, files)
-            }
-        }
-        return files.asSequence()
+        val (files, folders) = File(folder).listFiles().orEmpty().partition { it.isFile }
+//      Process files in the given folder
+        val inAndOutFiles = files.filter { it.isInFile() || it.isOutFile() }.groupBy { it.number() }
+        val inAndOutFilesMap = inAndOutFiles.toSortedMap().map { (number, files) ->
+            require(files.size == 2) { "There are less or more than 2 test files with number $number" }
```

Comment:
Do we guarantee this? In theory, I can have the following files: `(in_test_1.py, out_test_1.py)` and `(in_another_test_1.py, out_another_test_1.py)`. In this case, I have the same numbers. If we prohibit having such cases, let's describe it in the comments.

Reply:
Sorry, I just rewrote the functionality we already had and the examples you gave were not supported either in the previous version or in the current. Honestly, I don't see the way of supporting them since then we should pass the list of the input prefixes (in_test, in_another_test) and the list of the output prefixes, which seems so strange to implement. However, we can change our regex and allow some extra symbols between "in" and the number to consider all matched files as input, but shall we? Do we really need this functionality for tests?
But of course, we should add a comment (and maybe we should have done it earlier when we had much stricter restrictions)

Reply:
I think a comment will be enough